src/ai/backend/common/lock.py

```diff
@@ -44,6 +45,20 @@ async def __aenter__(self) -> Any:
     async def __aexit__(self, *exc_info) -> Optional[bool]:
         raise NotImplementedError
 
+    def enqueue_watchdog_task(self, ttl: float) -> None:
+        assert self._watchdog_task is None
+        self._watchdog_task = asyncio.create_task(self._watchdog_timer(ttl=ttl))
+
+    def dequeue_watchdog_task(self) -> None:
+        if task := self._watchdog_task:
+            if not task.done() and not task.cancelled():
+                task.cancel()
+        self._watchdog_task = None
+
+    @abc.abstractmethod
+    async def _watchdog_timer(self, ttl: float) -> None:
+        await asyncio.sleep(ttl)
```

Comment:
The concrete method of setting a TTL may not be represented as tasks depending on the backend implementation. In such case, we should be able to customize how we cancel the watchdog in subclasses. How could we abstract this as well?

Reply:
maybe we can pass a cancellation function as an argument to watchdog timer?